### DISCUSSION TITLE
Timeseries modifying line 'in_service' variable

### DISCUSSION THREAD

**Discussion by FrancisMujjuni** [e2nIEE/pandapower#1174] [Q&A]
At different time steps, the network I am working with has some lines in service and at other times, they are out of service. I have created a datasource of boolean variables for each line element at every timestep.  Unfortunately, the code cannot compile and my suspicion is that the DFData expects a numeric value rather than a boolean. I can get around this problem by simply looping over my datasource and changing the line 'in_service' value 'manually' to run the OPF but is there a simpler method in which I can implement the same using the run_timeseries module?

**Comment by rbolgaryn** [comment]
Hi @FrancisMujjuni ,

can you provide an example?

**Reply by rbolgaryn** [reply]
DataFrame should be able to have bool values
